src/main/java/org/deepsymmetry/beatlink/dbserver/ConnectionManager.java

```diff
@@ -299,6 +319,9 @@ private void requestPlayerDBServerPort(DeviceAnnouncement announcement) {
                 }
             }
         }
+        if(!portFound && retryCount > 0) {
+            requestPlayerDBServerPort(announcement, retryCount - 1);
+        }
```

Comment:
Since Java doesn’t have tail-call elimination, I will probably structure this as an ordinary loop, rather than nesting stack frames with recursive calls (not that there are likely to be that many of them I admit).